libs/cycle-finder/include/max_cycle.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include "core.hpp"
+#include "strongly_connected_components.hpp"
+#include <unordered_map>
+#include <vector>
+namespace cycleFinder
+{
+class MaxCycle {
+  private:
+    std::vector<std::vector<vertex>> cycles;
+    std::vector<std::vector<vertex>> maxCycles;
+    std::vector<vertex> stack;
+    std::set<vertex> blocked;
+    std::unordered_map<vertex, std::vector<vertex>> blockedMap;
+
+    core::multiGraph multiGraph;
+    StronglyConnectedComponents stronglyConnectedComponentsFinder;
+    vertex leastVertex;
```

Comment:
I opt for using `memberName_` case for class member variable. But like for n-th time in this review, this really should be done by clang format, not us.